lib/core/test/unit/Cardano/Wallet/Gen.hs

```diff
@@ -338,27 +328,15 @@ genScriptTemplateComplete :: Gen ScriptTemplate
 genScriptTemplateComplete = do
     script <- genScriptCosigners `suchThat` (not . null . retrieveAllCosigners)
     let scriptCosigners = retrieveAllCosigners script
-    xpubs <- vectorOf (length scriptCosigners) genXPub
+    xpubs <- vectorOf (length scriptCosigners) genMockXPub
     pure $ ScriptTemplate (Map.fromList $ zip scriptCosigners xpubs) script
 
-genRootKeySeqWithPass
-    :: Passphrase "encryption"
-    -> Gen (ShelleyKey depth XPrv)
-genRootKeySeqWithPass encryptionPass = do
-    s <- SomeMnemonic <$> genMnemonic @15
-    g <- Just . SomeMnemonic <$> genMnemonic @12
-    return $ Shelley.unsafeGenerateKeyFromSeed (s, g) encryptionPass
-
-genXPub :: Gen XPub
-genXPub = do
-    getRawKey . publicKey <$> (genRootKeySeqWithPass =<< genPassphrase (0, 16))
+genMockXPub :: Gen XPub
+genMockXPub = fromMaybe impossible . xpubFromBytes . BS.pack <$> genBytes
+  where
+    genBytes = vectorOf 64 arbitrary
+    impossible = error "incorrect length in genMockXPub"
```

Comment:
👍🏻

Reply:
yeah, we can generate here random bytes :100: